src/polybench/POLYBENCH_2MM-Hip.cpp

```diff
@@ -37,33 +57,60 @@ namespace polybench
   deallocHipDeviceData(C); \
   deallocHipDeviceData(D);
 
-
 __global__ void poly_2mm_1(Real_ptr tmp, Real_ptr A, Real_ptr B,
                            Real_type alpha,
-                           Index_type nj, Index_type nk)
+                           Index_type ni, Index_type nj, Index_type nk)
 {
-   Index_type i = blockIdx.y;
-   Index_type j = threadIdx.x;
-
-   POLYBENCH_2MM_BODY1;
-   for (Index_type k=0; k < nk; ++k) {
-     POLYBENCH_2MM_BODY2;
-   }
-   POLYBENCH_2MM_BODY3;
+  Index_type i = blockIdx.y * blockDim.y + threadIdx.y;
+  Index_type j = blockIdx.x * blockDim.x + threadIdx.x;
+
+  if ( i < ni && j < nj ) {
+    POLYBENCH_2MM_BODY1;
+    for (Index_type k=0; k < nk; ++k) {
+      POLYBENCH_2MM_BODY2;
+    }
+    POLYBENCH_2MM_BODY3;
+  }
+}
+
+template< typename Lambda >
+__global__ void poly_2mm_1_lam(Index_type ni, Index_type nj, Index_type nk,
```

Comment:
Looks like nk is unused here.

Reply:
you are correct sir. nk will be captured by the lambda for that. same with cuda. I will push a fix shortly.

Reply:
fixed.